source/dofs/dof_handler_policy.cc

```diff
@@ -2566,41 +2568,40 @@ namespace internal
                     numbers::artificial_subdomain_id)
                   {
                     // lines
-                    for (const auto l : cell->line_indices())
-                      {
-                        const auto line = cell->line(l);
-                        if (!line_touched[line->index()])
-                          {
-                            const unsigned int dofs_per_line =
-                              dof_handler.get_fe().n_dofs_per_line();
-                            dealii::types::global_dof_index *indices =
-                              &internal::DoFAccessorImplementation::
-                                Implementation::get_mg_dof_index(
-                                  dof_handler,
-                                  dof_handler.mg_levels[level],
-                                  dof_handler.mg_faces,
-                                  line->index(),
-                                  0,
-                                  0,
-                                  std::integral_constant<int, 1>());
-                            for (unsigned int d = 0; d < dofs_per_line; ++d)
-                              {
-                                if (check_validity)
-                                  Assert(indices[d] !=
-                                           numbers::invalid_dof_index,
-                                         ExcInternalError());
-
-                                if (indices[d] != numbers::invalid_dof_index)
-                                  indices[d] =
-                                    (indices_we_care_about.size() == 0) ?
-                                      new_numbers[indices[d]] :
-                                      new_numbers[indices_we_care_about
-                                                    .index_within_set(
-                                                      indices[d])];
-                              }
-                            line_touched[line->index()] = true;
-                          }
-                      }
+                    if (dofs_per_line > 0)
```

Comment:
You already checked for this in line 2555. I think one of these checks can go (either line 2555 or 2571).

Reply:
Unfortunately not, because the former is inside an `or` statement, so I need the latter. However, also the former has its use, because there is no need to iterate in case there are no dofs one quads or lines at all.

Reply:
Okay, thanks for the explanation :)